BUILD: Link ZLIB to CNPY

### Describe the bug

Currently, ZLIB is not linked to CNPY, which leads to a build error when compiling ABACUS with the `DENABLE_CNPY=1` option. The following error is encountered:
```bash
/usr/bin/ld: source/module_esolver/CMakeFiles/esolver.dir/io_npz.cpp.o: undefined reference to symbol 'crc32'
/usr/bin/ld: /lib/x86_64-linux-gnu/libz.so.1: error adding symbols: DSO missing from command line
icpx: error: linker command failed with exit code 1 (use -v to see invocation)

```

### Expected behavior

Modify the `CMakeLists.txt` to correctly link ZLIB with CNPY during the build process. This should resolve the undefined reference to symbol `crc32` error and ensure that the build completes successfully.

### To Reproduce

_No response_

### Environment

_No response_

### Additional Context

_No response_

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [ ] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -492,7 +492,10 @@ if (ENABLE_CNPY)
     include_directories(${cnpy_INCLUDE_DIR})
   endif()
   include_directories(${cnpy_SOURCE_DIR})
-  target_link_libraries(${ABACUS_BIN_NAME} cnpy)
+  
+  # find ZLIB and link
+  find_package(ZLIB REQUIRED)
+  target_link_libraries(${ABACUS_BIN_NAME} cnpy ZLIB::ZLIB)
   add_compile_definitions(__USECNPY)
 endif()
 

